docs/json_schemas/dns_record/v0/requirer.json

```diff
@@ -0,0 +1,227 @@
+        },
+        "dns_entries": {
+          "default": null,
+          "description": "List of DNS records for the provider to manage.",
+          "items": {
+            "$ref": "#/$defs/RequirerEntries"
+          },
+          "title": "Dns Entries",
+          "type": "array"
+        }
+      },
+      "title": "DNSRecordRequirer",
+      "type": "object"
+    },
+    "RecordClass": {
+      "const": "IN",
+      "description": "Represent the DNS record classes.",
+      "title": "RecordClass",
+      "type": "string"
+    },
+    "RecordType": {
+      "description": "Represent the DNS record types.",
+      "enum": [
+        "A",
+        "AAAA",
+        "CNAME",
+        "MX",
+        "DKIM",
+        "SPF",
+        "DMARC",
+        "TXT",
+        "CAA",
+        "SRV",
+        "SVCB",
+        "HTTPS",
+        "PTR",
+        "SOA",
+        "NS",
+        "DS",
+        "DNSKEY"
+      ],
+      "title": "RecordType",
+      "type": "string"
+    },
+    "RequirerDomains": {
+      "properties": {
+        "domain": {
+          "description": "Domain name for the provider to manage.",
+          "examples": [
+            "cloud.canonical.com",
+            "staging.ubuntu.com"
+          ],
+          "minLength": 1,
+          "name": "Domain",
+          "title": "Domain",
+          "type": "string"
+        },
+        "username": {
+          "description": "Username for authentication.",
+          "examples": [
+            "user1",
+            "user2"
+          ],
+          "name": "Username",
+          "title": "Username",
+          "type": "string"
+        },
+        "password": {
+          "description": "Juju secret containing the user password.",
+          "examples": [
+            "secret:123213123123123123123"
+          ],
+          "name": "Password",
+          "title": "Password",
+          "type": "string"
+        },
+        "uuid": {
+          "description": "UUID for this domain.",
+          "examples": "550e8400-e29b-41d4-a716-446655440000",
+          "minLength": 1,
+          "name": "UUID",
+          "title": "Uuid",
+          "type": "string"
+        }
+      },
+      "required": [
+        "domain",
+        "username",
+        "password",
+        "uuid"
+      ],
+      "title": "RequirerDomains",
+      "type": "object"
+    },
+    "RequirerEntries": {
+      "properties": {
+        "domain": {
+          "description": "Domain name requested.",
+          "examples": [
+            "cloud.canonical.com",
+            "staging.ubuntu.com"
+          ],
+          "minLength": 1,
+          "name": "Domain",
+          "title": "Domain",
+          "type": "string"
+        },
+        "host_label": {
+          "examples": [
+            "admin",
+            "www"
+          ],
+          "minLength": 1,
+          "name": "Host label",
+          "status_description": "Host label.",
+          "title": "Host Label",
+          "type": "string"
+        },
```

Comment:
I think we should fine tune the description because people may be confused by the `domain` and `host_label`
For example I want: `my.sub.domain.canonical.com`

What is the domain (canonical.com ? domain.canonical.com ? sub.domain.canonical.com ?) ?
And what is the host label ?

Bind is very flexible here and allows having a zone `sub.domain.canonical.com` that has a `my` DNS entry, or a zone `canonical.com` that has a `my.sub.domain` DNS entry

If the domain in dns-entries must absolutely match a domain in domain-entries we would also have that flexibility... but I'm worried people may be confused

Reply:
I think it's not a matter of being clearer... it's a matter of defining what is expected here.

Reply:
I believe the example already is as you mention @gregory-schiano

Reply:
I'm fine with that flexibility, but we'll have to document it clearly as the same DNS record could be describes in multiple ways:
For example `my.domain.tld` could be requested with:
- dns-domain: `domain.tld` and dns-record: `my`, which requires permission on `*.domain.tld`
- dns-domain: `my.domain.tld` and dns-record: `.`, which requires permission on `my.domain.tld`